src/CedarCore/SharedParameterManager.cs

```diff
@@ -82,9 +83,11 @@ public Definition RegisterParameter(string groupName, string parameterName, Para
                         new ExternalDefinitionCreationOptions(parameterName, parameterType);
                     newDefinition.UserModifiable = editable;
                     newDefinition.Description = description;
+                    newDefinition.GUID = id;
                     result = pilingGroup.Definitions.Create(newDefinition);
                 }
 
+                _application.SharedParametersFilename = currentPaath;
```

Comment:
Its not clear why you are having to read the initial file name, then change it do work and then reset it. 

Needs a comment explaining why.

Reply:
Its outlined in the api docs. The files contain a number of parameters, and rather than setting ours in the main system wide file we set it in a custom one for easy uninstall cleanup.

Reply:
I understand what is being having read the API docs in advance of the comment, however it's not clear why.  

From your response it just needs to something like the below;

"Rather than using main system wide file, setting parameters in a custom one for easy uninstall /cleanup.  Need to reset to main file once complete."

Reply:
Also, given your explanation you need to make sure that if the method fails for any reason _application.SharedParametersFilename is left in the correct state.